.circleci/build_image.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/sh
+set -xe
+
+if [[ $# -ne 2 ]]; then
+  echo "Usage: $0 DOCKERFILE IMAGE_NAME"
+  exit 2
+fi
+
+DIR=`dirname $1`
+DOCKERFILE=$1
+IMAGE_NAME=$2
+
+if [[ -z $CIRCLE_PULL_REQUEST ]]; then
+  # Log in to Docker Hub.
+  docker login -u $DOCKER_USER -p $DOCKER_PASS
```

Comment:
Also, I wonder if this `docker login` call couldn't also be moved down, just before the `docker tag`&`push` (assuming that we remove the `docker pull` call). That way, we'd only have a single `if [[ -z $CIRCLE_PULL_REQUEST ]]` block instead of two.